operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,304 @@
+	// metadata for the list
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+	// items is a slice of kueue
+	// this is a cluster scoped resource and there can only be 1 kueue
+	// +kubebuilder:validation:MaxItems=1
+	// +required
+	Items []Kueue `json:"items"`
+}
+
+// These structs come directly from Kueue.
+type Resources struct {
+	// excludeResourcePrefixes defines which resources should be ignored by Kueue
+	// +optional
+	// +kubebuilder:validation:items:MaxLength=64
+	// +kubebuilder:validation:MaxItems=16
+	ExcludeResourcePrefixes []string `json:"excludeResourcePrefixes,omitempty"`
+
+	// transformations defines how to transform PodSpec resources into Workload resource requests.
+	// This is intended to be a map with Input as the key (enforced by validation code)
+	// +optional
+	// +kubebuilder:validation:MaxItems=16
+	Transformations []ResourceTransformation `json:"transformations,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=Retain;Replace
+type ResourceTransformationStrategy string
+
+// ResourceTransformations apply transformation to pod spec resources
+// Retain means that we will keep the original resources and
+// apply a transformation.
+// Replace means that the original resources will be replaced
+// after the transformation is done.
+const Retain ResourceTransformationStrategy = "Retain"
+const Replace ResourceTransformationStrategy = "Replace"
+
+// ResourceTransformations apply transformation to pod spec resources
+// Retain means that we will keep the original resources and
+// apply a transformation.
+// Replace means that the original resources will be replaced
+// after the transformation is done.
+type ResourceTransformation struct {
+	// input is the name of the input resource.
+	// resources are pod spec resources like cpu, memory, gpus
+	// +required
+	Input corev1.ResourceName `json:"input"`
+
+	// strategy specifies if the input resource should be replaced or retained.
+	// retain means that we will keep the original resources and
+	// apply a transformation.
+	// replace means that the original resources will be replaced
+	// after the transformation is done.
+	// +optional
+	Strategy *ResourceTransformationStrategy `json:"strategy,omitempty"`
+
+	// outputs specifies the output resources and quantities per unit of input resource.
+	// An empty Outputs combined with a `Replace` Strategy causes the Input resource to be ignored by Kueue.
+	// +optional
+	Outputs corev1.ResourceList `json:"outputs,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=LessThanOrEqualToFinalShare;LessThanInitialShare
+type PreemptionStrategy string
+
+const (
+	LessThanOrEqualToFinalShare PreemptionStrategy = "LessThanOrEqualToFinalShare"
+	LessThanInitialShare        PreemptionStrategy = "LessThanInitialShare"
+)
+
+type FairSharing struct {
+	// enable indicates whether to enable fair sharing for all cohorts.
+	// this is disabled by default.
+	// +optional
+	Enable EnabledOrDisabled `json:"enable"`
```

Comment:
What does fair sharing actually do? Can you explain that in the godoc? Why as a user would I want to enable or disable this?